ignite/pkg/cosmosanalysis/app/app.go

```diff
@@ -125,56 +149,86 @@ func FindRegisteredModules(chainRoot string) ([]string, error) {
 		return true
 	})
 
-	return basicModules, nil
+	return modules, err
+}
+
+func exprToString(n ast.Expr) (string, error) {
+	buf := bytes.Buffer{}
+	fset := token.NewFileSet()
+
+	// Convert the expression node to Go
+	if err := format.Node(&buf, fset, n); err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func newExprError(msg string, n ast.Expr) error {
+	s, err := exprToString(n)
+	if err != nil {
+		return fmt.Errorf("%s: %w", msg, err)
+	}
+
+	return fmt.Errorf("%s: %s", msg, s)
 }
 
-func findBasicManagerRegistrations(n ast.Node, basicManagerModule string) []string {
+func findBasicManagerRegistrations(n ast.Node, basicManagerModule, pkgDir string, pkgs map[string]string) (packages []string, err error) {
 	callExprType, ok := n.(*ast.CallExpr)
 	if !ok {
-		return nil
+		return
 	}
 
 	selectorExprType, ok := callExprType.Fun.(*ast.SelectorExpr)
 	if !ok {
-		return nil
+		return
 	}
 
 	identExprType, ok := selectorExprType.X.(*ast.Ident)
 	if !ok || identExprType.Name != basicManagerModule || selectorExprType.Sel.Name != "NewBasicManager" {
-		return nil
+		return
 	}
 
-	packagesRegistered := make([]string, len(callExprType.Args))
-	for i, arg := range callExprType.Args {
-		argAsCompositeLitType, ok := arg.(*ast.CompositeLit)
-		if ok {
-			compositeTypeSelectorExpr, ok := argAsCompositeLitType.Type.(*ast.SelectorExpr)
-			if !ok {
-				continue
+	for _, arg := range callExprType.Args {
```

Comment:
```suggestion
+	// n is the call to NewBasicManager, let's loop on its args to discover modules
 	for _, arg := range callExprType.Args {
```

That doesn't hurt :)